src/main/java/org/kiwiproject/dropwizard/poller/ClientPoller.java

```diff
@@ -0,0 +1,492 @@
+    }
+
+    /**
+     * The URI the poller is polling. <em>Not required, but recommended</em>
+     * <p>
+     * NOTE: only used for logging purposes. It will be overwritten so we should not ever manually set it.
+     */
+    private URI uri;
+
+    /**
+     * A descriptive name for this poller. This will be used in logging and health checks to indicate which poller this is
+     */
+    @Getter
+    @Builder.Default
+    private String name = "Poller_" + System.currentTimeMillis();
+
+    /**
+     * Supplies the invoker used to make the repeating REST call. <em>Required</em>
+     */
+    private final Supplier<SyncInvoker> supplier;
+
+    /**
+     * The consumer that acts on the {@code Optional<Response>} from {@link SyncInvoker#get()}. <em>Required.</em>
+     */
+    private final Consumer<Response> consumer;
+
+    /**
+     * How the consumer should handle responses, sync or async. <em>Default is async</em,
+     */
+    @SuppressWarnings("UnusedAssignment")
+    @Builder.Default
+    private ConsumerType consumerType = ConsumerType.ASYNC;
+
+    /**
+     * The value after which a synchronous response consumer will time out.
+     */
+    @SuppressWarnings("UnusedAssignment")
+    @Getter(AccessLevel.PACKAGE)
+    @Builder.Default
+    private Long syncConsumerTimeout = DEFAULT_SYNC_RESPONSE_CONSUMER_TIMEOUT;
+
+    /**
+     * The unit of the synchronous response consumer time out value.
+     */
+    @SuppressWarnings("UnusedAssignment")
+    @Getter(AccessLevel.PACKAGE)
+    @Builder.Default
+    private TimeUnit syncConsumerTimeoutUnit = DEFAULT_SYNC_RESPONSE_CONSUMER_TIMEOUT_UNIT;
+
+    /**
+     * The value after which the {@link #supplier} will time out, e.g. if a poll is taking a very long time.
+     */
+    @SuppressWarnings("UnusedAssignment")
+    @Getter(AccessLevel.PACKAGE)
+    @Builder.Default
+    private Long supplierTimeout = DEFAULT_SUPPLIER_TIMEOUT;
+
+    /**
+     * The unit of the {@link #supplierTimeout}.
+     */
+    @SuppressWarnings("UnusedAssignment")
+    @Getter(AccessLevel.PACKAGE)
+    @Builder.Default
+    private TimeUnit supplierTimeoutUnit = DEFAULT_SUPPLIER_TIMEOUT_UNIT;
+
+    /**
+     * The delay before polling will begin.
+     */
+    @SuppressWarnings("UnusedAssignment")
+    @Builder.Default
+    private Duration initialExecutionDelay = DEFAULT_INITIAL_EXECUTION_DELAY;
+
+    /**
+     * The decision function that determines whether a poll should actually occur when the executor executes.
+     * <p>
+     * If not provided, a default decision function is used that always returns true.
+     *
+     * @implNote Suppress 54276 (Functional Interfaces should be as specialised as possible) because while the default
+     * {@link Function} doesn't use the {@link ClientPollerStatistics}, users might choose to use it, for example
+     * don't poll if we've had more than N failures in the last 5 minutes.
+     */
+    @SuppressWarnings({"UnusedAssignment", "squid:54276"})
```

Comment:
Check the Sonar number...if we keep it I think it would now be a "java" ☕️ not a "squid" 🦑

Reply:
Not an issue anymore, removed